src/mem/largealloc.h

```diff
@@ -145,14 +145,25 @@ namespace snmalloc
       {
         FlagLock f(lock);
 
-        auto aligned_bump = bits::align_up(bump, alignment);
-        if ((aligned_bump - bump) > remaining)
+        uint8_t* aligned_bump;
+        if constexpr (alignment != 0)
+        {
+          aligned_bump = pointer_align_up<alignment, uint8_t>(bump);
+        }
+        else
+        {
+          aligned_bump = static_cast<uint8_t*>(bump);
+        }
+
+        unsigned long bump_delta = static_cast<unsigned long>(

```

Comment:
Why is this `unsigned long` shouldn't this be `size_t`?  On 64 bit Windows long is only 32bit.

Reply:
`unsigned long` was me just being bad and placating my compiler, I'm sure.  `size_t` is the right answer and I'll move to that.